### PR TITLE
Support HTTPOption in Ingress spec

### PR DESCRIPTION
**Pull request by nak3**
This patch supports Ingress's `spec.HTTPOption`.

As a consequence, we can drop the dependency of config-network configmap.

Fix https://github.com/knative/networking/issues/303

/cc @dprotaso @kvmware

### CODE REVIEW

**Comment by nak3** [comment]
/retest

Not related failure.

**Comment by kvmware** [comment]
LGTM, Would want @dprotaso to do sign off though.

**Comment by dprotaso** [comment]
> As a consequence, we can drop the dependency of config-network configmap.

Is serving now setting the default value for `spec.HTTPOption`  based on the `config-network` setting?

**Comment by nak3** [comment]
Yes, it was done by https://github.com/knative/serving/commit/8179fc67dbb20144265a7ee0f77810b318b2aaca

File: pkg/reconciler/contour/lister.go [hunk 1]
```diff
@@ -44,15 +44,17 @@ func (l *lister) ListProbeTargets(ctx context.Context, ing *v1alpha1.Ingress) ([
...
-	}
-
 	visibilityKeys := cfg.Contour.VisibilityKeys
+
 	for key, hosts := range ingress.HostsPerVisibility(ing, visibilityKeys) {
+		port, scheme := int32(80), "http"
+
+		// Probe external servce with https.
+		if ing.Spec.HTTPOption == v1alpha1.HTTPOptionRedirected &&
+			!visibilityKeys["ClusterLocal"].Has(key) {
```
Comment (dprotaso) [inline pkg/reconciler/contour/lister.go:54]: cc @kvmware

I wonder if this line fixes this issue: https://github.com/knative-sandbox/net-contour/issues/423

**Review by dprotaso — APPROVED** [review]

**Pull request merged**